src/hmem_neuron.c

```diff
@@ -122,6 +122,41 @@ int neuron_copy_from_dev(uint64_t device, void *host, const void *dev, size_t si
 	return -FI_ENOSYS;
 }
 
+int neuron_get_handle(void *dev_buf, void **handle)
+{
+	return -FI_ENOSYS;
+}
```

Comment:
ofi_hmem.h has ENOSYS implementations that can be used instead.  E.g. ofi_hmem_no_get_handle

Reply:
The thing is that this function need to return 0 when there are neuron has been intialized. because shm calls it for all initialized hmem interfaces.

Reply:
I'm referring to all of the functions here that return ENOSYS.  We have existing ENOSYS calls.

The functions should not return success unless it actually succeeds.  This sounds like a bug in the shm implementation

Reply:
In the implementation
```c
```

Reply:
Sorry. Just realized that shm called `host_register` which should be a no-op for neuron if neuron is initialized.

shm did not call `get_handle`. 

@wenduwan I think we should continue use `ofi_hmem_no_get_handle` here.

Reply:
I see. Thanks I will open a separate PR to substitute those no op functions in both neuron and synapseai.